### PR TITLE
Feature #1440: support wildcards in rule filenames v1

### PR DESCRIPTION
**Pull request by gozzy**
Attempt to support wildcards in rule filenames. It's a draft, but seems to work. Any remarks or suggestions are welcome.

### CODE REVIEW

File: src/detect.c [hunk 1]
```diff
@@ -393,20 +439,14 @@ int SigLoadSignatures(DetectEngineCtx *de_ctx, char *sig_file, int sig_file_excl
...
-    int badfiles = 0;
-
-    int goodtotal = 0;
-    int badtotal = 0;
-
     char varname[128] = "rule-files";
+    int good_sigs = 0;
+    int bad_sigs = 0;
+
+    bzero(&sig_stat, sizeof(SigFileLoaderStat));
```
Comment (inliniac) [inline src/detect.c]: bzero is deprecated, please use memset

File: src/detect.c [hunk 2]
```diff
@@ -199,6 +199,8 @@
 
 #include "runmodes.h"
 
+#include <glob.h>
```
Comment (inliniac) [inline src/detect.c]: might be good to test for this header file in configure.ac

**Comment by inliniac** [comment]
If I read the code correctly this patch pushes all detect file loads through the new 'glob' function, where for exact matches it will work as before, and for patterns it loads multiple files. Correct?

**Comment by gozzy** [comment]
@inliniac, yes, for exact matches it works like it did before as glob() returns one match in this case.

File: src/detect.c (hunk 2 above)
Comment (gozzy) [inline src/detect.c]: Ok, will try to handle it. If header is not found, suricata should compile anyway, but without wildcard support?

Comment (inliniac) [inline src/detect.c]: Lets not bother with a 'not found'-path at first. I will run the new PR through my QA and if that passes we can assume glob.h is available everywhere.

**Comment by gozzy** [comment]
Replaced with #1500.

**Pull request closed without merging**